public/src/components/channelManagement/campaigns/StatusUpdateDialog.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  List,
+  ListItem,
+  IconButton,
+  makeStyles,
+  Switch,
+  Typography,
+} from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import { Test, Status } from '../helpers/shared';
+import { updateStatuses, FrontendSettingsType } from '../../../utils/requests';
+import { testChannelData } from './CampaignsEditor';
+
+const useStyles = makeStyles(() => ({
+  dialogHeader: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    paddingRight: '8px',
+  },
+  input: {
+    '& input': {
+      textTransform: 'uppercase !important',
+    },
+  },
+  onOffLabel: {
+    fontSize: '13px',
+  },
+  channelLabel: {
+    fontSize: '14px',
+    marginLeft: '16px',
+  },
+  testLabel: {
+    fontSize: '14px',
+    marginLeft: '10px',
+  },
+}));
+
+interface StatusUpdateDialogProps {
+  isOpen: boolean;
+  close: () => void;
+  tests: Test[];
+  updatePage: () => void;
+}
+
+interface TestStatus {
+  [index: string]: string;
+}
+
+type UpdateStatusesArguments = [FrontendSettingsType, string[], Status];
+
+const StatusUpdateDialog: React.FC<StatusUpdateDialogProps> = ({
+  isOpen,
+  close,
+  tests,
+  updatePage,
+}: StatusUpdateDialogProps) => {
+  const classes = useStyles();
+
+  const oldTestStatuses: TestStatus = {};
+
+  const [testData, setTestData] = useState<TestStatus>(oldTestStatuses);
+
+  const getStatusKey = (test: Test) => {
+    return `${test.channel}|${test.name}`;
+  };
+
+  useEffect(() => {
+    tests.map(test => {
+      const key = getStatusKey(test);
+      oldTestStatuses[key] = test.status;
+    });
+    setTestData(oldTestStatuses);
+  }, [tests]);
+
+  const updateSwitch = (e: React.ChangeEvent) => {
+    e.persist();
+
+    if (e.target != null) {
+      const key = e.target.id;
+      const newTestData = { ...testData };
+      newTestData[key] = testData[key] === 'Live' ? 'Draft' : 'Live';
+      setTestData(newTestData);
+    }
+  };
+
+  const onSubmit = (): void => {
+    const changes: UpdateStatusesArguments[] = [];
+    tests.forEach(test => {
+      const key = getStatusKey(test);
+      if (oldTestStatuses[key] !== testData[key]) {
+        const { channel, name } = test;
+        if (channel != null) {
+          const link = testChannelData[channel].link as FrontendSettingsType;
+          const data = testData[key] as Status;
+          changes.push([link, [name], data]);
+        }
+      }
+    });
+
+    if (changes.length > 0) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const promises: Promise<any>[] = [];
+      changes.forEach(change => {
+        promises.push(updateStatuses(...change));
+      });
+
+      Promise.all(promises).then(() => updatePage());
+    }
```

Comment:
You could group the names and only send one request per channel

Reply:
Agreed - it'll be much faster if you batch them, and the endpoint was designed to take a batch of test names.
It's a bit fiddly, but how about replacing `UpdateStatusesArguments` with something like this to group test names by channel/status:
```
interface StatusUpdates {
  [channel: string]: {
    [status in Status]: string[];
  }
}
```

Reply:
I'm happy to ship as it is for now, but I'll create a card to do batching later. We should leverage the backend/dynamo's ability to send a batch of updates as a single transaction. We can make the backend's endpoint more helpful here, so the client does less work.

Reply:
Hi @tomrf1  - what would your `happy place` be on this? My `happy place` would be a single endpoint which accepts an array of objects containing test update data (similar to the one you proposed above) which the backend could process and send out to Dynamo in a single update call.

Reply:
yes ideally a single endpoint which takes an array of objects with name/channel/status. Though currently the backend has a controller per channel so it may be a bit of a refactor. Or perhaps we can live with sending a request for each channel. Something to be decided when the card is picked up